### DISCUSSION TITLE
Database (and Server) Is Created Every Time

### DISCUSSION THREAD

**Discussion by Mike-E-angelo** [microsoft/aspire#5474] [Q&A] [answered]
Hi Community,

I am investigating Aspire, and am having moderate luck.  Everything seems to be working OK.  However, one aspect I just learned is that during local development it appears the database + server (container) are created new each and every time, erasing the old server + database altogether.

This can be problematic. :)

I am new to containers and this world.  Is there a way to keep the container retaining the data that it had from the previous local development run?

Thank you for any assistance you can provide.

**Comment by ryanheath** [comment, marked as answer]
Try something like this, here we use sqlserver.

```
IResourceBuilder<SqlServerServerResource> sqlServer;
if (builder.ExecutionContext.IsPublishMode)
{
    sqlServer = builder.AddSqlServer(SqlServerName);
}
else
{
    sqlServer = builder.AddSqlServer(SqlServerName)
        .WithDataVolume();
}
```

// Ryan

**Reply by davidfowl** [reply]
Right you can use volumes to persist data. There's a method `WithDataVolume` that will configure various containers to persist data in a docker (or podman) volume. This will make sure the data doesn't go away between runs.

**Reply by Mike-E-angelo** [reply]
Thank you very much for the additional context! 🙏  I have suggested updating documentation to reduce this sort of confusion for future users here:
https://github.com/dotnet/docs-aspire/issues/1574
